src/main/java/lrbg/codriver/data/TaskList.java

```diff
@@ -0,0 +1,170 @@
+package lrbg.codriver.data;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+import lrbg.codriver.data.exception.CoDriverException;
+import lrbg.codriver.parser.Parser;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    /** The list of tasks. */
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor for TaskList.
+     * @param taskStrings The array of strings representing tasks.
+     */
+    public TaskList(String[] taskStrings) {
+        this.tasks = new ArrayList<>();
+        for (String s : taskStrings) {
+            String[] arguments = s.split("\\|");
+            String type = arguments[0];
+            boolean isDone = arguments[1].equals("1");
+            switch (type) {
+            case "T": // task.ToDo
+                String description = arguments[2];
+                Todo t = new Todo(description);
+                if (isDone) {
+                    t.markDone();
+                }
+                this.tasks.add(t);
+                break;
+            case "D": // task.Deadline
+                String deadlineDescription = arguments[2];
+                LocalDate deadlineDate = Parser.parseDate(arguments[3]);
+                Deadline d = new Deadline(deadlineDescription, deadlineDate);
+                if (isDone) {
+                    d.markDone();
+                }
+                this.tasks.add(d);
+                break;
+            case "E": // task.Event
+                String eventDescription = arguments[2];
+                String[] dates = arguments[3].split("~");
+                LocalDate eventFrom = Parser.parseDate(dates[0]);
+                LocalDate eventTo = Parser.parseDate(dates[1]);
+                Event e = new Event(eventDescription, eventFrom, eventTo);
+                if (isDone) {
+                    e.markDone();
+                }
+                this.tasks.add(e);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Gets the size of the list.
+     * @return The size of the list.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Gets the task at the given index.
+     * @param index The index of the task, zero indexed.
+     * @return The task at the given index.
+     */
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
```

Comment:
In addition to the comment on the `size()` method, would it be better to rename this method as `getTask(int index)`?